src/main/java/com/yahoo/memory/BaseWritableBufferImpl.java

```diff
@@ -111,16 +111,17 @@ public void getByteArray(final byte[] dstArray, final int dstOffset, final int l
 
   //OTHER PRIMITIVE READ METHODS: copyTo, compareTo XXX
   @Override
-  public int compareTo(final long thisOffsetBytes, final long thisLengthBytes, final Buffer that,
+  public int compareTo(final long thisOffsetBytes, final long thisLengthBytes, final Buffer thatBuf,
           final long thatOffsetBytes, final long thatLengthBytes) {
     state.checkValid();
-    that.getResourceState().checkValid();
     checkBounds(thisOffsetBytes, thisLengthBytes, capacity);
+    final BaseWritableBufferImpl that = (BaseWritableBufferImpl) thatBuf;
+    that.state.checkValid();
     checkBounds(thatOffsetBytes, thatLengthBytes, that.capacity);
     final long thisAdd = getCumulativeOffset() + thisOffsetBytes;
     final long thatAdd = that.getCumulativeOffset() + thatOffsetBytes;
-    final Object thisObj = (isDirect()) ? null : unsafeObj;
-    final Object thatObj = (that.isDirect()) ? null : ((WritableBuffer)that).getArray();
+    @SuppressWarnings("UnnecessaryLocalVariable") final Object thisObj = this.unsafeObj;
```

Comment:
My IDE declares that these @SuppressWarnings("UnnecessaryLocalVariable") are unsupported.  Never seen them before.  And I don't agree with warning's claim that these vars are unnecessary.  Dereferencing these before the loop makes all kinds of sense and we can't rely on the compiler or JIT to do it.